types/eslint/index.d.ts

```diff
@@ -1401,12 +1458,15 @@ export namespace ESLint {
 
         // Autofix
         fix?: boolean | ((message: Linter.LintMessage) => boolean) | undefined;
-        fixTypes?: Array<Rule.RuleMetaData["type"]> | undefined;
+        fixTypes?: Array<"directive" | "problem" | "suggestion" | "layout"> | undefined;
```

Comment:
> `"directive" | "problem" | "suggestion" | "layout"`

[Feature] Similar to the `CodePathOrigin`, this union of string literals shows up twice. It'd be good to add something like:

```ts
type FixType = "directive" | "problem" | "suggestion" | "layout";
```

There are a few other unions of string literals in this file. I'll hold off commenting on all of them individually.

Reply:
Thanks! I've added these union types:

* `Rule.CodeOriginPath`
* `ESLint.FixType`
* `ESLint.CacheStrategy`
* `Linter.EcmaVersion`
* `Linter.SourceType`

I believe these are all unions of nontrivial types that are used in multiple places, but let me know if I missed something.